### PR TITLE
PigLatin: Update to use templates

### LINKED ISSUE #1969: Pig Latin: Triggering on empty input
See [this query](https://duckduckgo.com/?q=pig+latin&t=lm&ia=answer).

screenshot from 2015-12-27 17 19 45

-----
IA Page: https://duck.co/ia/view/pig_latin

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
Updates the [PigLatin IA](https://duck.co/ia/view/pig_latin) to using the `text` text answer template.

Also Fixes #1969 with a simple trigger check.

-----
IA Page: https://duck.co/ia/view/pig_latin

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata post coming soon

**Comment by moollaza** [comment]
Deploying to beta...

**Comment by GuiltyDolphin** [comment]
@moollaza Done a bunch of fixes/improvements to make it more awesome! (Now supports bidirectional translations - *mostly*).

**Comment by GuiltyDolphin** [comment]
@moollaza This is hopefully all done - ready for review.

File: lib/DDG/Goodie/PigLatin.pm [hunk 1]
```diff
@@ -3,23 +3,51 @@ package DDG::Goodie::PigLatin;
...
+    $action = 'to' if $action eq 'in';
+    return ($query, $action);
+}
 
-    return "Pig Latin: " . $out,
-      structured_answer => {
-        input     => [html_enc($in)],
-        operation => 'Translate to Pig Latin',
-        result    => html_enc($out)
+handle query_raw => sub {
```
Comment (moollaza) [inline lib/DDG/Goodie/PigLatin.pm]: fwiw `handle query_lc` is probably better to work with. Whitespace is normalized, and casing is lowered, which should make for simpler regexes

File: lib/DDG/Goodie/PigLatin.pm [hunk 2]
```diff
@@ -3,23 +3,51 @@ package DDG::Goodie::PigLatin;
...
+    my $result = $action eq 'to'
+        ? to_piglatin($to_translate)
+        : from_piglatin($to_translate);
+    return if $result eq $to_translate;
+
+    return $result, structured_answer => {
+        id   => 'pig_latin',
+        name => 'Answer',
+        data => {
+            title    => "$result",
```
Comment (moollaza) [inline lib/DDG/Goodie/PigLatin.pm]: you shouldn't need the quotes, but it doesn't hurt.

**Comment by moollaza** [comment]
@GuiltyDolphin this is looking good. I think we are probably going to want to reduce the trigger space thought because most queries of the form `piglatin foo` are not trying to convert something to piglatin. 

The triggers "to piglatin", "in piglating", "translate to piglatin" are much more indicative of a translation search. For example the top autocomplete queries for "pig latin" all result in unnecessary translations:

e.g. https://duckduckgo.com/?q=pig+latin+translator&t=ffab&ia=answer

File: lib/DDG/Goodie/PigLatin.pm (hunk 2 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/PigLatin.pm]: @moollaza I kindah prefer to just have quotes around anything that is being passed around as text - makes it a bit easier to reason about it.

**Comment by GuiltyDolphin** [comment]
@moollaza Sure! Most of my changes dealt with making it so you could say `X in piglatin` etc... I'll remove the imperative version.

File: lib/DDG/Goodie/PigLatin.pm (hunk 1 above)
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/PigLatin.pm]: @moollaza I need to be able to handle casing, which is why I haven't used `query_lc`.

Comment (moollaza) [inline lib/DDG/Goodie/PigLatin.pm]: Whoops, my bad :+1: 

If you use `query` then at least the whitespace will be normalized

Comment (GuiltyDolphin) [inline lib/DDG/Goodie/PigLatin.pm]: @moollaza Nice. Didn't spot that - thanks.

**Comment by moollaza** [comment]
@GuiltyDolphin Thanks! This LGTM :+1:

**Pull request merged**